pkg/yurthub/proxy/util/util.go

```diff
@@ -301,6 +304,60 @@ func WithRequestTimeout(handler http.Handler) http.Handler {
 	})
 }
 
+func WithNonResourceRequest(handler http.Handler, cfg *config.YurtHubConfiguration, rest *rest.RestConfigManager) http.Handler {
```

Comment:
I think the normal usage for `Withxxx` is to add external info to req.Context(). We'd better not to handle the request during these functions.